situp-plugins/apmtracesource/src/main/java/com/amazon/situp/plugins/processor/oteltrace/OTelTraceRawProcessor.java

```diff
@@ -150,13 +61,24 @@ private static String convertStringNanosToISO8601(final String stringNanos) {
         final List<Record<String>> finalRecords = new LinkedList<>();
         for(Record<ExportTraceServiceRequest> ets: records) {
             for (ResourceSpans rs : ets.getData().getResourceSpansList()) {
-                String jsonString;
                 try {
-                    jsonString = getJsonFromProtobufObj(rs);
-                    decodeResourceSpan(jsonString).forEach(doc -> finalRecords.add(new Record<>(doc)));
-                } catch (InvalidProtocolBufferException | JsonProcessingException e) {
-                    log.warn("Unable to process invalid records", e);
+                    final String serviceName = OTelProtoHelper.getServiceName(rs.getResource()).orElse(null);
+                    final Map<String, Object> resourceAttributes = OTelProtoHelper.getResourceAttributes(rs.getResource());
+                    for (InstrumentationLibrarySpans is : rs.getInstrumentationLibrarySpansList()) {
+                        for (Span sp : is.getSpansList()) {
+                            try {
+                                finalRecords.add(new Record<>(new RawSpanBuilder()
+                                        .setFromSpan(sp, is.getInstrumentationLibrary(), serviceName, resourceAttributes)
+                                        .build().toJson()));
+                            } catch (Exception ex) {
+                                log.error("Unable to process invalid Span {}:", sp, ex);
```

Comment:
Question: JFI: Were we doing the same before as well - if the processor fails to process even a single span, we will fail ?

Reply:
So the request object contains 1 or more resourceSpans and each resourceSpan has 1 or more spans. We will skip the failed and move to next in both case. There is no reason for retry as it will not work.